File: MissingNumber.java
import java.util.Scanner;

class MissingNumber {
    public static void main(final String[] args) {
        final Scanner scan = new Scanner(System.in);
        final int num = scan.nextInt();
        final int[] arr = new int[num];
        int sum = 0;
        for (int i = 0; i < arr.length - 1; i++) {
            arr[i] = scan.nextInt();
        }
        sum = num * (num + 1) / 2;
        for (int i = 0; i < arr.length; i++) {
            sum -= arr[i];
        }
        System.out.println(sum);

        scan.close();
    }
}

File: missingNumber.cpp
#include <bits/stdc++.h>
using namespace std;

void missingNumber(int arr[], int size)
{
    int sum = (size * (size + 1)) / 2;
    int missingNumber = 0;
    sort(arr, arr + size);
    for (int i = 0; i < size; i++)
    {
        sum -= arr[i];
        if (arr[i] == arr[i + 1])
        {
            i++;
            missingNumber = arr[i];
        }
    }
    cout << sum << " " << missingNumber << endl;
}

int main()
{
    int arr[] = {1, 2, 3, 4, 1};
    int size = 5;
    missingNumber(arr, size);
}